### ISSUE TITLE
Different Behavior Between Windows and Linux

### ISSUE THREAD

**Issue by FaithBeam** [TolikPylypchuk/SharpHook#20]
Hello! Great library! I'd like to use this library as an autowalker in games. I press a button on my mouse and w or shift + w is sent to move my character forward in game. I've noticed two issues when trying to accomplish this. 

1. On Windows, in a game where shift represents sprinting, shift + w doesn't engage sprint. My character will move forward, but it is like shift isn't being held down. On Linux, the shift + w combo works as expected where the character will sprint forward.
2. On Windows, the event simulator seems spotty as to whether or not it works in game. Some games work and others not. I can see the mouse hook working in all of them because it fires the OnMousePressed event, but the event simulator SimulateKeyPress does nothing. On Linux there seems to be no issue using it this way in the same games.

This is the code I tested with. .Net 6, using top level statements, SharpHook 3.1.1.

```Csharp
using SharpHook;
using SharpHook.Native;

var hook = new TaskPoolGlobalHook();
var simulator = new EventSimulator();
var state = false;

hook.MousePressed += OnMousePressed;
await hook.RunAsync();

void OnMousePressed(object? sender, MouseHookEventArgs mouseHookEventArgs)
{
    if (mouseHookEventArgs.Data.Button != MouseButton.Button4) return;
    if (state)
    {
        Console.WriteLine("Released");
        simulator.SimulateKeyRelease(KeyCode.VcLeftShift);
        simulator.SimulateKeyRelease(KeyCode.VcW);
        state = false;
    }
    else
    {
        Console.WriteLine("Pressed");
        simulator.SimulateKeyPress(KeyCode.VcW);
        simulator.SimulateKeyPress(KeyCode.VcLeftShift);
        state = true;
    }
}
```

**Comment by TolikPylypchuk** [comment]
Hi! Thanks for using this library!

As for the first issue, I've just tested this code by running it and simply clicking around in Visual Studio and watching which character will be printed. With your code I got `w` which means that it was indeed simulating pressing just `w`, without `shift`. But the thing is that we need to press `shift` before we press `w` for it to be counted as `shift+w`. So, when I did this instead it worked as expected and printed `W`:

```c#
simulator.SimulateKeyPress(KeyCode.VcLeftShift);
simulator.SimulateKeyPress(KeyCode.VcW);
```

As for the second issue, I can't really tell as I'm not a game developer nor am I a gamer. I think it may be because some games use anti-cheat systems which suppress simulated events, but that's just a guess.

**Comment by TolikPylypchuk** [comment]
This looks OK, but I don't have much knowledge about the specifics for sending events on Windows. Did you test your changes? If they work for you i.e., the games can now see the event, and it doesn't look like it breaks anything, then I'm down to add this change.

**Comment by FaithBeam** [comment]
I don't have much experience on that either. I have tested this change and it does work in game. Here is a video showing it. In the video I have a test c# project with the fixed code you pointed out with SharpHook 3.1.1 installed. I test it against notepad to show it enters W (shift+w), and then in the game Half Life 2: Lost Coast. I then change the SharpHook 3.1.1 nupkg to 3.1.2 (my local version with only the change in the libuiohook library) and do the same steps to show the difference.

https://www.youtube.com/watch?v=jg8FKPygiXY

Since we're both not sure, I will create an issue in upstream libuiohook because they probably know this much better than we do. Upstream also has the same issue. If they think it's good, then it should be safe to update your fork with the change.

**Comment by TolikPylypchuk** [comment]
Yeah, you can do that, but bear in mind that the last commit in libuiohook was 6 months ago, so it's not in active development at the moment, and I'm not sure when your change would be merged.

I think it's safe enough to add to SharpHook, and if it turns out that it does break something, then we can always revert it. I'll add it to my fork of libuiohook in the coming days, and then release version 3.1.2.

**Comment by FaithBeam** [comment]
That's unfortunate but if someone else has that problem, hopefully they stumble into the issue. Here it is, for the record https://github.com/kwhat/libuiohook/issues/157

That would be great, thank you!

**Comment by TolikPylypchuk** [comment]
I've released version 3.1.2 with [the change that you suggested](https://github.com/TolikPylypchuk/libuiohook/commit/fc779b0bc892f8aaf373b53a3791f1e5590b9924), so this issue can be closed :)
